homeassistant/components/hue/hue_event.py

```diff
@@ -0,0 +1,101 @@
+"""Representation of a Hue remote firing events for button presses."""
+import logging
+
+from aiohue.sensors import TYPE_ZGP_SWITCH, TYPE_ZLL_ROTARY, TYPE_ZLL_SWITCH
+
+from homeassistant.const import CONF_EVENT, CONF_ID
+from homeassistant.core import callback
+from homeassistant.util import slugify
+
+from .sensor_device import GenericHueDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_HUE_EVENT = "hue_event"
+CONF_LAST_UPDATED = "last_updated"
+CONF_UNIQUE_ID = "unique_id"
+
+EVENT_NAME_FORMAT = "{}"
+
+
+class HueEvent(GenericHueDevice):
+    """When you want signals instead of entities.
+
+    Stateless sensors such as remotes are expected to generate an event
+    instead of a sensor entity in hass.
+    """
+
+    def __init__(self, sensor, name, bridge, primary_sensor=None):
+        """Register callback that will be used for signals."""
+        super().__init__(sensor, name, bridge, primary_sensor)
+
+        self.event_id = slugify(self.sensor.name)
+        # Use the 'lastupdated' string to detect new remote presses
+        self._last_updated = self.sensor.lastupdated
+
+        # Register callback in coordinator and add job to remove it on bridge reset.
+        self.bridge.sensor_manager.coordinator.async_add_listener(
+            self.async_update_callback
+        )
+        self.bridge.reset_jobs.append(self.async_will_remove_from_hass)
+        _LOGGER.debug("Hue event created: %s", self.event_id)
+
+    @callback
+    def async_will_remove_from_hass(self):
+        """Remove listener on bridge reset."""
+        self.bridge.sensor_manager.coordinator.async_remove_listener(
+            self.async_update_callback
+        )
+
+    @callback
+    def async_update_callback(self):
+        """Fire the event if reason is that state is updated."""
+        if self.sensor.lastupdated == self._last_updated:
+            return
+
+        # Extract the press code as state
+        if hasattr(self.sensor, "buttonevent"):
+            state = self.sensor.buttonevent
+        elif hasattr(self.sensor, "rotaryevent"):
+            state = self.sensor.rotaryevent
+        else:
+            state = self.sensor.state
+
+        self._last_updated = self.sensor.lastupdated
+
+        # Fire event
+        data = {
+            CONF_ID: self.event_id,
+            CONF_UNIQUE_ID: self.unique_id,
```

Comment:
What's the difference between both ?

Reply:
* The first one (the _slugified_ one) is a user-given, friendly name, like "kitchen_remote". 
* The second one is the ugly mac/id of the device (`"00:aa:bb:cc:dd:ee:ff:00"`)

It is **like it's done in deCONZ**, and has its logic, as the pretty name is more useful to be used in automations and doesn't reveal any real identifier from the devices.